MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -121,11 +121,15 @@ public class SimulatedLocationManager: NavigationLocationManager {
         let distance = min(max(distanceToClosest, 10), safeDistance)
         let coordinatesNearby = polyline.trimmed(from: newCoordinate, distance: 100).coordinates
         
-        // Simulate expected speed
-        if let speeds = routeProgress?.currentLeg.segmentSpeeds,
-            let closestCoordinateOnRoute = Polyline(routeProgress!.route.coordinates!).closestCoordinate(to: lookAheadCoordinate),
-            closestCoordinateOnRoute.index < speeds.count {
-            currentSpeed = speeds[closestCoordinateOnRoute.index]
+        // Simulate speed based on expected segment travel time
+        if let expectectedSegmentTravelTimes = routeProgress?.currentLeg.expectedSegmentTravelTimes,
+            let coordinates = routeProgress?.route.coordinates,
+            let closestCoordinateOnRoute = Polyline(routeProgress!.route.coordinates!).closestCoordinate(to: newCoordinate),
+            let nextCoordinateOnRoute = coordinates.after(element: coordinates[closestCoordinateOnRoute.index]),
+            let time = expectectedSegmentTravelTimes.optional[closestCoordinateOnRoute.index] {
+            
+            let distance = coordinates[closestCoordinateOnRoute.index].distance(to: nextCoordinateOnRoute)
+            currentSpeed = distance / time
```

Comment:
Any idea why this leads to choppier simulation than the speed attribute? I thought this is the very definition of the API’s speed calculation.

Reply:
Anyways, if it’s a problem, we can switch to the other suggestion in https://github.com/mapbox/mapbox-navigation-ios/pull/683#discussion_r142565325, which is to have ViewController add the speed attribute when the simulation switch is on (and document that developers should do likewise when enabling simulation).

Reply:
I'm seeing roughly the same results. @bsudekum, can you try again using a device or a different route?